report_compassion/models/res_partner.py

```diff
@@ -8,9 +8,11 @@
 #    The licence is in the file __manifest__.py
 #
 ##############################################################################
-from datetime import date
+from locale import setlocale
```

Comment:
This was wrong. You should import setlocale function from there
`from odoo.addons.thankyou_letters.models.res_partner import setlocale`